### PR TITLE
generate our own sa token signer when missing

### PR DESCRIPTION
**Pull request by deads2k**
Instead of using the installer created SA token signer, create a new one.  This relies on https://github.com/openshift/origin/pull/21971 and https://github.com/openshift/cluster-kube-apiserver-operator/pull/238

/assign @sttts

### CODE REVIEW

**Comment by deads2k** [comment]
/hold

this needs to wait until the endpoints list shows the bootstrap kube-apiserver (which won't respect it) gone

**Comment by deads2k** [comment]
/hold cancel

@sttts new commit added

File: pkg/operator/certrotationcontroller/satokensigner_controller.go [hunk 1]
```diff
@@ -8,6 +8,8 @@ import (
 	"fmt"
 	"time"
 
+	"k8s.io/apimachinery/pkg/util/sets"
```
Comment (sttts) [inline pkg/operator/certrotationcontroller/satokensigner_controller.go]: nit: order

Comment (deads2k) [reply pkg/operator/certrotationcontroller/satokensigner_controller.go]: got it

File: pkg/operator/certrotationcontroller/satokensigner_controller.go [hunk 2]
```diff
@@ -91,6 +98,45 @@ func (c SATokenSignerController) sync() error {
...
+		for _, subset := range kubeEndpoints.Subsets {
+			for _, address := range subset.Addresses {
+				if !nodeIPs.Has(address.IP) {
+					unexpectedEndpoints.Insert(address.IP)
+				}
+			}
+		}
+		if len(unexpectedEndpoints) != 0 {
+			message := fmt.Sprintf("unexpected addresses:", unexpectedEndpoints.List())
+			c.eventRecorder.Warning("SATokenSignerControllerStuck", message)
```
Comment (sttts) [inline pkg/operator/certrotationcontroller/satokensigner_controller.go]: is this really a warning? Should be normal during bootstrapping

**Comment by sttts** [comment]
Looks good. Only https://github.com/openshift/cluster-kube-controller-manager-operator/pull/153#discussion_r257285104.

File: pkg/operator/certrotationcontroller/satokensigner_controller.go (hunk 2 above)
Comment (deads2k) [reply pkg/operator/certrotationcontroller/satokensigner_controller.go]: fixed

**Comment by deads2k** [comment]
/hold

flaky as all get out.

**Comment by deads2k** [comment]
/hold cancel

there is now a staging position until they pass the time allotted.  We can fix that later by having a better signal, but the code now tolerates waiting for that signal.

**Comment by deads2k** [comment]
/hold 

holding for debug output

**Comment by deads2k** [comment]
/retest

network

**Comment by deads2k** [comment]
network

/retest

**Comment by deads2k** [comment]
known flake on bootstrap user.  other failures were known networking flakes.

**Pull request merged**